### ISSUE TITLE
eval_beamsearch_ngram.py fails (pack_hypotheses invalid data type 'str')

### ISSUE THREAD

**Issue by devvindan** [NVIDIA/NeMo#6127]
**Describe the bug**

I trained QuartzNet acoustic model + KenLM 2-ngram model, now I run eval_beamsearch_ngram.py to evaluate it. 

It seems like beam_search_func in BeamSearchDecoderWithLM outputs tuples (confidence, text) (the doc agrees), so cand.y_sequence is a resulting string. 

but later on the function pack_hypothesis in ctc_beam_decoding does "cand.y_sequence = torch.tensor(cand.y_sequence, dtype=torch.long)" and expects it to be int labels, so I get an error. Meanwhile greedy decoding works perfectly.

**Steps/Code to reproduce bug**

I am just following the steps here https://docs.nvidia.com/deeplearning/nemo/user-guide/docs/en/stable/asr/asr_language_modeling.html

python eval_beamsearch_ngram.py nemo_model_file=*quartznet_model_from_default_config* \
           input_manifest="" \
           kenlm_model_file=*scorer from train_kenlm* \
           beam_width=[500] \
           beam_alpha=[0.82] \
           beam_beta=[0.87] \
           preds_output_folder= \
           probs_cache_file=null \
           decoding_mode=beamsearch_ngram

**Expected behavior**

I expected the eval_beamsearch_ngram.py script work with default QuartzNet and default KenLM trained model. I am not exactly sure if I am not specifying the correct params for it. 

**Environment overview (please complete the following information)**

 - Environment location: linux
 - Method of NeMo install: pip install from tutorial

**Environment details**

Otherwise, please provide:
- OS version: ubuntu 18.04.5 LTS (Bionic Beaver)
- PyTorch version 1.13.1
- Python version 3.8.12

**Additional context**

nemo toolkit 1.16.0rc0
ctc-decoders version is 1.1

**Comment by titu1994** [comment]
Hmm I didn't test this with QuartzNet but it seems with char based model it is not correctly returning a list of integer IDs but a str. I will take a look next week
